Rename plugin

qreshi opened the pull request:
Signed-off-by: Mohammad Qureshi <qreshi@amazon.com>

### Description

Updating plugin name to contain `opensearch`
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

dblock commented on build.gradle:
```diff
@@ -87,7 +87,7 @@ testClusters.javaRestTest {
 allprojects {
     group = 'com.amazon.opendistroforelasticsearch'
-    version = "${opendistroVersion}.0"
+    version = "${opendistroVersion}.0-beta1"
```
Let's get rid of these `opendistroVersion` while we're at it?

dblock commented on build.gradle:
```diff
@@ -71,7 +71,7 @@ validateNebulaPom.enabled = false
 opensearchplugin {
-    name 'opendistro-job-scheduler'
+    name 'opensearch-job-scheduler'
     description 'Open Distro for Elasticsearch job schduler plugin'
```
Let's fix the description to be OpenSearch, including the `schduler` typo?

qreshi commented:
@dblock Addressed the comments.

I also had a couple questions.

1. What should we update this to in `build.gradle`?:
https://github.com/opensearch-project/job-scheduler/blob/0918a568d215d2b448cd21fcd350f20d38a1c7ed/build.gradle#L140-L146

2. Previously ODFE (OpenDistro for Elasticsearch) used to refer to the distribution whereas Elasticsearch referred to the backend search engine. Now, OpenSearch refers to both the distribution and the backend search engine. Do you think this could get confusing at all when referencing distribution versions/engine versions and the like?

dblock commented:
> @dblock Addressed the comments.
> 
> I also had a couple questions.
> 
> 1. What should we update this to in `build.gradle`?:
>    https://github.com/opensearch-project/job-scheduler/blob/0918a568d215d2b448cd21fcd350f20d38a1c7ed/build.gradle#L140-L146
> 2. Previously ODFE (OpenDistro for Elasticsearch) used to refer to the distribution whereas Elasticsearch referred to the backend search engine. Now, OpenSearch refers to both the distribution and the backend search engine. Do you think this could get confusing at all when referencing distribution versions/engine versions and the like?

I'd be guessing on both. I think we should standardize this soon,  /cc:@saratvemulapalli

saratvemulapalli commented:
> > @dblock Addressed the comments.
> > I also had a couple questions.
> > 
> > 1. What should we update this to in `build.gradle`?:
> >    https://github.com/opensearch-project/job-scheduler/blob/0918a568d215d2b448cd21fcd350f20d38a1c7ed/build.gradle#L140-L146
> > 2. Previously ODFE (OpenDistro for Elasticsearch) used to refer to the distribution whereas Elasticsearch referred to the backend search engine. Now, OpenSearch refers to both the distribution and the backend search engine. Do you think this could get confusing at all when referencing distribution versions/engine versions and the like?
> 
> I'd be guessing on both. I think we should standardize this soon, /cc:@saratvemulapalli

Yeah definitely a great question. There are multiple questions involved, which are open:

1. Url for downloads
2. Docs URL
3. Distinguishing between backend engine version and plugin version.
